include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -177,8 +177,27 @@ class binary_writer
 
             case value_t::number_float:
             {
-                oa->write_character(get_cbor_float_prefix(j.m_value.number_float));
-                write_number(j.m_value.number_float);
+		const auto valueD = j.m_value.number_float;
+		const float valueF = static_cast<float>(valueD);
+		if (std::isnan(valueD)) {
+		    // NaN is 0xf97e00 in CBOR
+		    oa->write_character(to_char_type(0xF9));
+		    oa->write_character(to_char_type(0x7E));
+		    oa->write_character(to_char_type(0x00));
+		} else if (std::isinf(valueD)) {
+		    // Infinity is 0xf97c00, -Infinity is 0xf9fc00
+		    oa->write_character(to_char_type(0xf9));
+		    oa->write_character(valueD > 0 ? to_char_type(0x7C) : to_char_type(0xFC));
+		    oa->write_character(to_char_type(0x00));
+		} else {
+                    if(static_cast<double>(valueF) == valueD) {
```

Comment:
Please run `make amalgamate` to execute astyle on the code.

Reply:
OK, thanks, I'll do this according to `CONTRIBUTING.md`